app/controllers/api/v2/access_requests_controller.rb

```diff
@@ -1,19 +1,28 @@
 module API
   module V2
     class AccessRequestsController < API::V2::ApplicationController
-      before_action :build_access_request
+      before_action :build_access_request, except: :list
+      before_action :build_list_of_access_requests, except: :approve
 
       def approve
         result = AccessRequestApprovalService.call(@access_request)
 
         render status: 200, json: { result: result }
       end
 
+      def list
+        render status: 200, json: @list_of_access_requests
+      end
+
     private
 
       def build_access_request
         @access_request = authorize AccessRequest.requested.find(params[:id])
       end
+
+      def build_list_of_access_requests
+        @list_of_access_requests = authorize AccessRequest.requested.all
```

Comment:
This instance variable should be `@access_requests` and you shouldn’t need the `all`

Reply:
I think it'd be better to split this into two lines:

```ruby
authorize AccessRequest
@access_requests = AccessRequest.requested
```

The first line is a slightly different way to do the authorisation because lists of things are authorised slightly differently to individual things. See `API::V2::ProvidersController#index` and `API::V2::CoursesController#index` that follow the pattern of authorising the class, rather than the collection that you get back from ActiveRecord.

Reply:
What's funny about this is i initially had it as @access_requests, but changed it due to Heidar's voice popping into my head saying it wasn't very clear. Have actioned this. Thanks for the feedback.